pkg/context/context.go

```diff
@@ -444,3 +445,19 @@ func (c *Context) SetSensitiveValues(vals []string) {
 		c.Err = err
 	}
 }
+
+// UserAgent returns a string that can be used as a user agent for porter.
+func (c *Context) UserAgent() string {
+	product := "porter"
+
+	if pkg.Commit == "" && pkg.Version == "" {
+		return product
+	}
+
+	v := pkg.Version
+	if len(v) == 0 {
```

Comment:
You don't have to change this logic here but I wanted you to know that version unless you run a build using `go build` manually, all the build scripts always set both a version and a commit. The version will be the output of running `git describe` so that even untagged commits can express a version value.